### PR TITLE
Report total folder size as MB, GB, or TB

### PR DESCRIPTION
**Pull request by ramonsmits**
Alternative to #2480 to report the sizes as MB, GB, or TB.

[ByteSize 2.0.0 license is MIT](https://github.com/omar/ByteSize/blob/v2.0.0/LICENSE)

Example of startup info:

```txt
-------------------------------------------------------------
ServiceControl Version:             4.17.0-report-foldersize-humanize.1+56.Branch.report-foldersize-humanize.Sha.0a74daee389ee6283d319f77c56b69916b6534fa
Audit Retention Period (optional):
Error Retention Period:             10.00:00:00
Ingest Error Messages:              True
Forwarding Error Messages:          False
Database Size:                      1.05 MB
Database Folder Size:               119.93 MB
ServiceControl Logging Level:       Info
RavenDB Logging Level:              Warn
Selected Transport Customization:   ServiceControl.Transports.Learning.LearningTransportCustomization, ServiceControl.Transports.Learning
-------------------------------------------------------------
```

### CODE REVIEW

File: src/ServiceControl/ServiceControl.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
     <ProjectReference Include="..\ServiceControl.Infrastructure.RavenDB\ServiceControl.Infrastructure.RavenDB.csproj" />
     <ProjectReference Include="..\ServiceControl.LicenseManagement\ServiceControl.LicenseManagement.csproj" />
     <ProjectReference Include="..\ServiceControl.SagaAudit\ServiceControl.SagaAudit.csproj" />
+    <ProjectReference Include="..\ServiceControl.Transports.Learning\ServiceControl.Transports.Learning.csproj" />
```
Comment (danielmarbach) [inline src/ServiceControl/ServiceControl.csproj:15]: Should not be here

File: src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs [hunk 2]
```diff
@@ -168,15 +170,48 @@ long DataSize()
...
+        {
+            try
+            {
+                var dir = new DirectoryInfo(settings.DbPath);
+                var dirSize = DirSize(dir);
+                return dirSize;
+            }
+            catch
+            {
+                return -1;
```
Comment (danielmarbach) [inline src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs:183]: What happens when we return `-1` and pass it through humanizer?

Comment (ramonsmits) [reply src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs:183]: @danielmarbach Good question :-)

**Review by danielmarbach — APPROVED** [review]
Other than the comment LGTM

**Comment by danielmarbach** [comment]
Given that Humanizer uses MIT License and seems to still be fairly active I think it would be OK to take a dependency on it.

File: src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs (hunk 2 above)
Comment (ramonsmits) [reply src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs:183]: @danielmarbach `-1 B` :-)

Comment (danielmarbach) [reply src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs:183]: Seems to work https://dotnetfiddle.net/hJZtd7

**Comment by ramonsmits** [comment]
@danielmarbach Replaces the package with a smaller one. We could also instead just copy that file into the solution instead of taking a package dependency.

**Review by danielmarbach — APPROVED** [review]

File: src/ServiceControl/Bootstrapper.cs [hunk 3]
```diff
@@ -174,26 +176,58 @@ long DataSize()
...
             var startupMessage = $@"
 -------------------------------------------------------------
 ServiceControl Version:             {version}
 Audit Retention Period (optional):  {settings.AuditRetentionPeriod}
 Error Retention Period:             {settings.ErrorRetentionPeriod}
 Ingest Error Messages:              {settings.IngestErrorMessages}
 Forwarding Error Messages:          {settings.ForwardErrorMessages}
-Database Size:                      {DataSize()} bytes
+Database Size:                      {dataSize:n0} bytes ({ByteSize.FromBytes(dataSize).ToString("#.##", CultureInfo.InvariantCulture)})
+Database Folder Size:               {folderSize:n0} bytes ({ByteSize.FromBytes(folderSize).ToString("#.##", CultureInfo.InvariantCulture)})
```
Comment (danielmarbach) [inline src/ServiceControl/Bootstrapper.cs:230]:
```suggestion
Database Size:                      {ByteSize.FromBytes(dataSize).ToString("#.##", CultureInfo.InvariantCulture)}
Database Folder Size:               {ByteSize.FromBytes(folderSize).ToString("#.##", CultureInfo.InvariantCulture)}
```

File: src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs [hunk 4]
```diff
@@ -168,15 +170,48 @@ long DataSize()
...
+            var dataSize = DataSize();
+            var folderSize = FolderSize();
             var startupMessage = $@"
 -------------------------------------------------------------
 ServiceControl Audit Version:       {version}
 Audit Retention Period:             {settings.AuditRetentionPeriod}
 Forwarding Audit Messages:          {settings.ForwardAuditMessages}
-Database Size:                      {DataSize()} bytes
+Database Size:                      {dataSize:n0} bytes ({ByteSize.FromBytes(dataSize).ToString("#.##", CultureInfo.InvariantCulture)})
+Database Folder Size:               {folderSize:n0} bytes ({ByteSize.FromBytes(folderSize).ToString("#.##", CultureInfo.InvariantCulture)})
```
Comment (danielmarbach) [inline src/ServiceControl.Audit/Infrastructure/Bootstrapper.cs:214]:
```suggestion
Database Size:                      {ByteSize.FromBytes(dataSize).ToString("#.##", CultureInfo.InvariantCulture)}
Database Folder Size:               {ByteSize.FromBytes(folderSize).ToString("#.##", CultureInfo.InvariantCulture)}
```

**Review by danielmarbach — COMMENTED** [review]
Now that I saw the output I think there is little value in presenting the bytes

**Review by danielmarbach — APPROVED** [review]

**Comment by danielmarbach** [comment]
@dependabot merge

**Pull request merged**